### PR TITLE
[NFC] Use clang::Type->dump() to account for upstream changes.

### PR DESCRIPTION
**Pull request by varungandhi-apple**
Fixes rdar://65302410.

master-rebranch version of https://github.com/apple/swift/pull/33019

### CODE REVIEW

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test

**Review by nathawes — APPROVED** [review]
Thanks Varun! Don't worry about the test failures. The branch was cut before a bunch of the master-next fixes got in. I've got a PR up at the moment to fix that and will retest and merge this once that lands.

**Pull request merged**
